File: Algorithms/bubble_sort.py

user_input = input("Enter a list of numbers in any order (space separated) : ")
input_list = [int(i) for i in user_input.strip().split(' ')]


# The loop will should iterate (n-1) times. After the first round, the largest element will be 
# present at the end of the list
def bubble_sort(l, si, ei):
	for j in range(0, ei):
		for i in range(0, ei):
			if l[i] > l[i+1]:
				l[i], l[i+1] = l[i+1], l[i]		# Swapping if adjacent elements are not in order


bubble_sort(input_list, 0, len(input_list) - 1)
print(input_list)

File: DS_with_C++/searching_and_sorting_challenges/bubble_sort.cpp
#include<iostream>
using namespace std;

void bubbleSort(int a[], int n)
{
	for (int i = 0; i < n; ++i)
	{
		for (int j = 0; j < n - i - 1; ++j)
		{
			if (a[j] >  a[j + 1])
				swap(a[j], a[j + 1]);
		}
	}
}

int main() {

	int n;
	cin >> n;		// No. of elements in the array
	int a[n];

	for (int i = 0; i < n; ++i)
		cin >> a[i];
	
	bubbleSort(a, n);

	for (int i = 0; i < n; ++i)
		cout << a[i] << endl;

	return 0;
}
